### ISSUE TITLE
Previewer not working in my project

### ISSUE THREAD

**Issue by alex6dj** [ForNeVeR/AvaloniaRider#228]
Even using solutions from other Issues, I hope you see anything I cant see.
My project link: https://github.com/QAppsSoft/MultiConverter

If more info nedded, I will do my best.

Thanks in advance.

Rider: v2022.2
AvaloniaRider: v1.0.0.2140933520
Avalonia: v11 preview 1
Windows: 10 1903

Log:
```
Socket error while session is still alive

java.net.SocketException: Connection reset
	at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:323)
	at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:350)
	at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:803)
	at java.base/java.net.Socket$SocketInputStream.read(Socket.java:966)
	at java.base/java.io.DataInputStream.readFully(DataInputStream.java:201)
	at java.base/java.io.DataInputStream.readFully(DataInputStream.java:172)
	at me.fornever.avaloniarider.bson.BsonStreamReader.readBytes(BsonStreamReader.kt:26)
	at me.fornever.avaloniarider.bson.BsonStreamReader.readMessage(BsonStreamReader.kt:30)
	at me.fornever.avaloniarider.previewer.AvaloniaPreviewerSession.processSocketMessages(AvaloniaPreviewerSession.kt:64)
	at me.fornever.avaloniarider.previewer.AvaloniaPreviewerSessionController$executePreviewerAsync$sessionJob$1.invokeSuspend(AvaloniaPreviewerSessionController.kt:299)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by ForNeVeR** [comment]
The previewer logs say the following:
```
Unhandled exception. System.InvalidOperationException: Failed to resolve object of type MultiConverter.ViewModels.MainViewModel
   at MultiConverter.DependencyInjection.ReadonlyDependencyResolverExtensions.GetRequiredService[TService](IReadonlyDependencyResolver resolver) in C:\Temp\MultiConverter\src\MultiConverter\DependencyInjection\ReadonlyDependencyResolverExtensions.cs:line 13
   at MultiConverter.App.GetRequiredService[T]() in C:\Temp\MultiConverter\src\MultiConverter\App.axaml.cs:line 40
   at MultiConverter.App.OnFrameworkInitializationCompleted() in C:\Temp\MultiConverter\src\MultiConverter\App.axaml.cs:line 20
   at Avalonia.Controls.AppBuilderBase`1.Setup() in /_/src/Avalonia.Controls/AppBuilderBase.cs:line 289
   at Avalonia.Controls.AppBuilderBase`1.SetupWithoutStarting() in /_/src/Avalonia.Controls/AppBuilderBase.cs:line 136
   at Avalonia.DesignerSupport.Remote.RemoteDesignerEntryPoint.AppInitializer`1.ConfigureApp(IAvaloniaRemoteTransportConnection transport, CommandLineArgs args, Object obj) in /_/src/Avalonia.DesignerSupport/Remote/RemoteDesignerEntryPoint.cs:line 154
   at Avalonia.DesignerSupport.Remote.RemoteDesignerEntryPoint.Main(String[] cmdline) in /_/src/Avalonia.DesignerSupport/Remote/RemoteDesignerEntryPoint.cs:line 179
   at Avalonia.Designer.HostApp.Program.Main(String[] args) in /_/src/tools/Avalonia.Designer.HostApp/Program.cs:line 47
```

This is a problem of your program, something's impossible to resolve.

**Comment by ForNeVeR** [comment]
Likely, the reason is that these dependencies are registered in your `Main` method, and not `BuildAvaloniaApp`.

The previewer only calls `BuildAvaloniaApp`, and so it's unable to get your dependencies.

**Comment by alex6dj** [comment]
If you can tell my where the AvaloniaRider logs are located

**Comment by alex6dj** [comment]
> Likely, the reason is that these dependencies are registered in your `Main` method, and not `BuildAvaloniaApp`.
> 
> The previewer only calls `BuildAvaloniaApp`, and so it's unable to get your dependencies.

Ok, solved, finally.
Dependencies loading moved to `BuildAvaloniaApp()`. I didnt catch that. Any way, I found the logs but are hard to follow mixed with all the others, even if you know what to search for.

Thanks again, this previewer is a life (Time) saver for Styles.

Get my :star2:

**Comment by ForNeVeR** [comment]
FYI, the logs are shown in the IDE, in the Avalonia tool window. But it is known they aren't very discoverable this way, and I plan to improve that before preparing a public release of the plugin.
